### PR TITLE
debounced the button reading

### PR DESCRIPTION
**Pull request by parzival0152**

### CODE REVIEW

File: morse_module/morse.c [hunk 1]
```diff
@@ -198,16 +168,44 @@ int main() {
 		bool tx  = gpio_get(TX);
 
 		// make sure that the button isnt bounding
+		inc_count = inc ? inc_count + 1 : 0;
+		dec_count = dec ? dec_count + 1 : 0;
+		tx_count = tx ? tx_count + 1 : 0;
 
 		// handle the press of the button
+		
+		if(inc_count > 5) {
```
Comment (Mipsters) [inline morse_module/morse.c:177]: constify the shit

File: morse_module/morse.c [hunk 2]
```diff
@@ -198,16 +168,44 @@ int main() {
...
+		}
 
 		if(fail_flag) { // If there was a FAIL, hold the LED red for a bit and then go back to normal.
 			gpio_put(RED, 1);
 			sleep_ms(500);
 			gpio_put(RED, 0);
 			fail_flag = 0;
 		}
-		sleep_ms(25);
+		sleep_ms(20);
```
Comment (Mipsters) [inline morse_module/morse.c:208]: more constify

**Review by Mipsters — CHANGES_REQUESTED** [review]

File: morse_module/morse.c (hunk 1 above)
Comment (parzival0152) [reply morse_module/morse.c]: consted

File: morse_module/morse.c [hunk 3]
```diff
@@ -42,6 +42,9 @@ static uint8_t skip_press = 0;
 const uint I2C_SLAVE_ADDRESS = 0x18;
 const uint SCREEN_ADDR = 0x27;
 
+const uint8_t debounce_factor = 5;
```
Comment (Mipsters) [inline morse_module/morse.c:45]: uppercase

**Review by Mipsters — CHANGES_REQUESTED** [review]

**Review by Mipsters — APPROVED** [review]

**Pull request merged**